crypto/rand/randfile.c

```diff
@@ -286,8 +286,22 @@ const char *RAND_file_name(char *buf, size_t size)
         if (OPENSSL_strlcpy(buf, s, size) >= size)
             return NULL;
     } else {
+#ifdef OPENSSL_SYS_WINDOWS
+        /*
+         * We use the same env variables as GetTempFile() - but that function
+         * uses TCHARs, but getenv() gives us chars so its easier to do it this
+         * way
```

Comment:
I sort of agree. Well I agree that it would be best to handle this properly. Perhaps it would have been better to put a "TODO" in the comment. I didn't see it as the scope of this MR to fix it though. We were already using getenv() so in that sense this commit doesn't make anything worse (and actually makes things better). Fixing the UTF8 support is a whole different bug that just happens to hit the same bit of code.